### ISSUE TITLE
Cannot create public IP addresses when using AZs

### ISSUE THREAD

**Issue by dsboulder** [cloudfoundry/bosh-azure-cpi-release#608]
### Versions (*please add them if you have more releases for this issue*)
CPI      Version: all
BOSH     Version:
CF       Version:
Diego    Version:

### Stemcell
Version:
Distro:
  - [x] Ubuntu
  - [ ] CentOS

### Environment
- Azure Environment:
  - [x] Azure
  - [ ] AzureChinaCloud
  - [ ] AzureUSGovernment
  - [ ] AzureGermanCloud
  - [ ] AzureStack
    - Version: (e.g. 1712 update or build number 180106.1)
  - [ ] Others:
- Location: (e.g. East US)
- Subscription ID:
- Resource Group Name:
- [ ] Use Managed Disks

### Issue Details
When you enable AZs, it requires VMs to use Standard SKU Public IPs, but the CPI only creates Basic SKU IPs.

### Logs (*Please redact all sensitive information in your logs*)

```
Error message: {
  "error": {
    "code": "ZonesNotAllowedInBasicSkuResource",
    "message": "Zones not allowed in basic SKU resource /subscriptions/52641c3e-894d-443e-b78b-e00305c8e021/resourceGroups/pcf-2-5-azure-clean-pcf-resource-group-34672af3/providers/Microsoft.Network/publicIPAddresses/c974bc95-1d24-44ee-b1e9-4349f1b47abe in this region. Please use standard SKU.",
    "details": []
  }
}
```

**Comment by dsboulder** [comment]
Question: should make a global CPI option or just make the all IPs be standard public IPs.

**Comment by dsboulder** [comment]
Suggested logic:

- If AZ cloud property is set, use standard LB
- If LB is provided, go look at SKU and match SKU
- else use basic IP

Above logic is safe for patch and requires no operator input.

**Comment by bingosummer** [comment]
Do you mean the dynamic public IP? CPI should create a standard IP and LB if AZ is set. I'm surprised that it doesn't. Do you have the CPI logs or steps to reproduce it? Thanks!

**Comment by dsboulder** [comment]
It seems as if VMs on AZs do not use standard SKU IPs. Maybe we missed that criteria as well?

**Comment by dsboulder** [comment]
Fixed on 35.4.x and master.
